test/langtools/tools/javac/parser/ExtraPackageSemicolon.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8305671
+ * @summary Verify an extra semicolon is allowed after package decl with no imports
+ * @modules jdk.compiler/com.sun.tools.javac.code
+*/
+
+import com.sun.tools.javac.Main;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class ExtraPackageSemicolon {
+
+    public static void runTest(String filename, String source) throws Exception {
+        final File sourceFile = new File(filename);
+        System.err.println("writing: " + sourceFile);
+        try (PrintStream output = new PrintStream(new FileOutputStream(sourceFile))) {
+            output.println(source);
+        }
+        final StringWriter diags = new StringWriter();
+        final String[] params = new String[] { sourceFile.toString() };
+        System.err.println("compiling: " + sourceFile);
+        int ret = Main.compile(params, new PrintWriter(diags, true));
+        System.err.println("exit value: " + ret);
+        String output = diags.toString().trim();
+        if (!output.isEmpty())
+            System.err.println("output:\n" + output);
+        else
+            System.err.println("no output");
+        if (ret != 0)
+            throw new AssertionError("compilation failed, but expected success");
+    }
+
+    public static void main(String... args) throws Exception {
+        runTest("Test1.java", "package p;");
+        runTest("Test2.java", "package p;;");
+        runTest("Test3.java", "package p;; ;; ;; ;;; ;;; ;;; ;;");
+    }
```

Comment:
I suggest including test cases where the package statement and semicolons are followed by something else, like an import declaration or a type declaration.

Reply:
I'll add more tests, but extra semicolons prior to import statements aren't allowed; see [JDK-8027682].

Reply:
> I'll add more tests, but extra semicolons prior to import statements aren't allowed;

That would make it a negative test, and you would check that an appropriate error is generated.
That being said, the test case might be included in the tests for [JDK-8027682].

Reply:
I note there are no tests that *list* 8027682 in their list of bugs.
I note that there is a test included in the patch for 8027682, but it lists an unrelated bug [JDK-5059679].
It's also a minimal test that does not include test cases for extra semicolons between a package statement and the first import statement.

Reply:
> > I'll add more tests, but extra semicolons prior to import statements aren't allowed;
> 
> That would make it a negative test, and you would check that an appropriate error is generated. That being said, the test case might be included in the tests for [JDK-8027682].

Gotcha. Checking for proper detection of illegal extra semicolons is already covered in [the fix for JDK-8027682 fix](https://github.com/openjdk/jdk/commit/4b8f7db6be80e425bebfaf6f68d49da74f29386a).

Reply:
> I note there are no tests that _list_ 8027682 in their list of bugs. If a test is designed to test the fix for a bug, it should list the bug number in the `@bug` line in the test description. I note that there is a test included in the patch for 8027682, but it lists an unrelated bug [JDK-5059679].

Ah yes - that's a bug bug (so to speak). Must have been a cut & paste error on my part.

I've filed [JDK-8309511](https://bugs.openjdk.org/browse/JDK-8309511) to fix that.